target/linux/ramips/dts/mt7628an_xiaomi_mi-router-4a-100m-intl-v2.dts

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "mt7628an_xiaomi_mi-router-4.dtsi"
+
+/ {
+	compatible = "xiaomi,mi-router-4a-100m-intl-v2", "mediatek,mt7628an-soc";
+	model = "Xiaomi Mi Router 4A (100M International Edition V2)";
+
+	aliases {
+		led-boot = &led_power_yellow;
+		led-failsafe = &led_power_yellow;
+		led-running = &led_power_blue;
+		led-upgrade = &led_power_yellow;
+		label-mac-device = &ethernet;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power_blue: power_blue {
+			label = "blue:power";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_yellow: power_yellow {
+			label = "yellow:power";
+			gpios = <&gpio 44 GPIO_ACTIVE_LOW>;
+		};
+
+		wan {
+			label = "blue:wan";
+			gpios = <&gpio 37 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+};
+
+&partitions {
+	partition@60000 {
+		label = "overlay";
+		reg = <0x60000 0x100000>;
+		read-only;
+	};
+
+	partition@160000 {
+		label = "firmware";
+		reg = <0x160000 0xea0000>;
+		compatible = "denx,uimage";
+	};
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		compatible = "mediatek,mt76";
+		reg = <0x0000 0 0 0 0>;
+		nvmem-cells = <&eeprom_factory_8000>;
```

Comment:
The MT7613 eeprom size is `0x4da8`. Please override it. An example: https://github.com/openwrt/openwrt/commit/ffa40fcc7fe1aee041775e097bccecb9be83cd21

Reply:
Unfortunatly I see comment in wr902-v4: `/* **tl-wr902ac-v4 has** different eeprom size '0x4da8' for MT7613 */`
But I cannot found confirmation for r4acv2 device. There are [some dmesg](https://forum.openwrt.org/t/xiaomi-4a-r4acv2-new-version-bricked/166933/2) on forum with factory partition size 10000. Moreover, I can't found confirmation about MT7613's eeprom. It is an radio chip, not SoC. Where did you found this information?

Reply:
This value is calculated from `mt76` driver. All mt7613/mt7615 have the same eeprom size. I believe you can only get very low tx power (about 3 dbm) now due to wrong eeprom size. see https://github.com/openwrt/openwrt/pull/13587

Reply:
Thank you very much!
I've did some tests and looked into factory's hexdump. I found that at `0x8000` begins new block with `0x600` bytes length. Space from `0x8600` up to end (`0x100000`) filled by `0xff`. I tried to use `0x600` and `0x4000`, but it gave me same result: only 3dBm.
But when I used all available partition (`0x100000` - `0x080000` = `0x80000`) I've got working wireless. Is there any reason to use exactly `0x4da8` bytes? If you may, tell me, how did u found this magic size?

Reply:
> But when I used all available partition (`0x100000` - `0x080000` = `0x80000`) I've got working wireless. Is there any reason to use exactly `0x4da8` bytes? If you may, tell me, how did u found this magic size?

You can add some debug codes to dump the expected eeprom length `len` or directly analyze the `mt76` source code.
https://github.com/openwrt/mt76/blob/bebd9cffc2aeb2cecb40aadbb8c6eab3bdf7971b/eeprom.c#L129-L132

> As I can understand old-style `mediatek,mtd-eeprom = <&factory 0x8000>;` supplied rest of `factory` partition.

No. The driver only reads the necessary length.

Reply:
> directly analyze the mt76 source code.

Thank you very much!
mt7615/eeprom.h says:
len = `MT7615_EEPROM_TXDPD_OFFSET +  MT7615_EEPROM_TXDPD_COUNT *  MT7615_EEPROM_TXDPD_SIZE`
where `...TXDPD_OFFSET = MT7615_EEPROM_SIZE (1024 (in mt7615.h)) + MT7615_EEPROM_DCOC_COUNT (34) * MT7615_EEPROM_DCOC_SIZE (256)` and `..TXDPD_COUNT = 44 + 3 = 47` and `...TXDPD_SIZE = 216`. Altogether it will be 19880 or 0x4da8.
Now it's still a magic, but I can accept this explanation) You're awesome!)